File: backend/ControleGastos.API/ControleGastos.API/Models/Transacao.cs
namespace ControleGastos.API.Models
{
    public class Transacao
    {
        public int Id { get; set; } // representa o identificador único da transação / pk no banco de dados
        public string Descricao { get; set; } // representa a descrição da transação
       public decimal Valor { get; set; } // representa o valor da transação
       public TipoTransacao Tipo { get; set; } // representa o tipo da transação (entrada ou saída)
        public DateTime Data { get; set; } // representa a data da transação
        public int PessoaId { get; set; } // representa o identificador da pessoa 
        public Pessoa Pessoa { get; set; } = null!;
    }
}

File: frontend/src/models/Transacao.ts
// representa os tipos possíveis de transação

export type TipoTransacao = 1 | 2;


// representa uma transação financeira
// possui os mesmos campos retornados pela api

export interface Transacao {

    // id da transação no banco de dados
    id: number;

    // descrição informada no cadastro
    descricao: string;

    // valor financeiro da movimentação
    valor: number;

    // tipo da transação
    // 1 = receita
    // 2 = despesa
    tipo: TipoTransacao;

    // data da transação
    data: string;

    // id da pessoa relacionada
    pessoaId: number;

    // nome da pessoa relacionada
    nomePessoa?: string;

}